crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -1167,8 +1201,14 @@ impl Loader<HistoricalKey> for Db {
                 continue;
             }
 
-            let object =
-                Object::try_from_stored_history_object(stored.clone(), key.checkpoint_viewed_at)?;
+            let object = Object::try_from_stored_history_object(
+                stored.clone(),
+                key.checkpoint_viewed_at,
+                // This will infer the `Object::root_version` if it's not a child object, but if it
+                // is, it will be set to `None`, so dynamic object queries will be bounded by
+                // `checkpoint_viewed_at`.
```

Comment:
```suggestion
                // This conversion will use the object's own version as the `Object::root_version` if it's not a child object, but if it
                // is, it will be set to `None`, since we do not have any information on the `root_version` of the fetched dynamic object.
```

Reply:
This makes me wonder if we should explicitly `infer_root_version` instead of deferring it to the function(s)?